agent/lib/broker_address_settings.js

```diff
@@ -18,9 +18,21 @@
 var log = require("./log.js").logger();
 var kubernetes = require('./kubernetes.js');
 var myutils = require('./utils.js');
+var events = require('events');
+var util = require('util');
 
 function kubernetes_resource_compare(a, b) {
-    return myutils.string_compare(a.metadata.name, b.metadata.name);
+    return myutils.string_compare(a.name, b.name);
+}
+
+function extract_spec(def) {
```

Comment:
name does not reflect the implementation well.  How about something like flatten_plan?